MediaProcessor/src/FFmpegConfigManager.cpp

```diff
@@ -77,12 +139,106 @@ VideoCodec FFmpegConfigManager::getVideoCodec() const {
     return m_videoSettings.codec;
 }
 
-std::unordered_map<AudioCodec, std::string>& FFmpegConfigManager::getAudioCodecAsString() {
+// Value Map as String Getters
+const std::unordered_map<AudioCodec, std::string>& FFmpegConfigManager::getAudioCodecAsString()
+    const {
     return m_audioCodecAsString;
 }
 
-std::unordered_map<VideoCodec, std::string>& FFmpegConfigManager::getVideoCodecAsString() {
+const std::unordered_map<VideoCodec, std::string>& FFmpegConfigManager::getVideoCodecAsString()
+    const {
     return m_videoCodecAsString;
 }
 
+const std::unordered_map<CodecStrictness, std::string>&
+FFmpegConfigManager::getCodecStrictnessAsString() const {
+    return m_codecStrictnessAsString;
+}
+
+// Update Settings
+int FFmpegConfigManager::updateSettings(
+    int globalBitField = 0, int audioBitField = 0, int videoBitField = 0, bool overwrite = false,
+    CodecStrictness strictness = CodecStrictness::EXPERIMENTAL, const fs::path& inputFilePath = "",
+    const fs::path& outputFilePath = "", AudioCodec audioCodec = AudioCodec::AAC,
+    const fs::path& chunksPath = "", const std::vector<fs::path>& processedChunkCol = {},
+    int chunkIndex = 0, int audioSampleRate = 0, int audioChannels = 0, double audioStartTime = 0,
+    double audioDuration = 0, double overlapDuration = 0,
+    VideoCodec videoCodec = VideoCodec::H264) {
+    int result = 0;
+
+    // Global Updates
+    if (globalBitField & GlobalConfigMask::Overwite) {
+        m_globalSettings.overwrite = overwrite;
+        result |= GlobalConfigMask::Overwite;
+    }
+
+    if (globalBitField & GlobalConfigMask::Strictness) {
+        m_globalSettings.strictness = strictness;
+        result |= GlobalConfigMask::Strictness;
+    }
+
+    if (globalBitField & GlobalConfigMask::InputFilePath) {
+        m_globalSettings.inputFilePath = inputFilePath;
+        result |= GlobalConfigMask::InputFilePath;
+    }
+
+    if (globalBitField & GlobalConfigMask::OutputFilePath) {
+        m_globalSettings.outputFilePath = outputFilePath;
+        result |= GlobalConfigMask::OutputFilePath;
+    }
```

Comment:
I know it was me who proposed the idea of this "check for changed fields", but I feel it's difficult to justify it. We're going to add all this code just so we can update only the changed settings... If we were to make a full update like a `PUT`, it'd be so simple:

```cpp
m_globalSettings = globalValues;  // just assign the new struct
```
and we'd be done. 

I've spent some time to see how we can avoid passing a bunch of args and to avoid cascaded contional checks. 
We could do it with a new helper class and some template functions, but at that stage it really feels like overkill for a relatively cheap operation. 

I really appreciate you taking this on, but I think going for a full update here would keep things simple and maintainable, but please let me know what you think!

Reply:
Hi @omeryusufyagci, no worries, it is definitely a lot of extra code! I have a few thoughts on how do do this, and I'm curious what you prefer. I suppose it depends on how "full" a full update should be.

1. Keep it simple with overloads, one version of updateSettings() for each FFmpeg settings struct
2. Use std::optional so the user can pass 1-2 structs without overwriting the third
3. Require a **full** update, user must pass 3 new structs

If you had something else in mind let me know, that's just how I'm reading it right now. Thanks for the discord link too! I don't spend a lot of time on there but I'll join the channel :)